test/livebook/runtime/node_pool_test.exs

```diff
@@ -0,0 +1,69 @@
+defmodule Livebook.Runtime.NodePoolTest do
+  use ExUnit.Case, async: true
+
+  alias Livebook.Runtime.NodePool
+
+  # Tests for Livebook.Runtime.NodePool
+  #
+  # Note:
+  # 1. We do not spawn actual nodes as it can be time
+  #    intensive (on low spec machines)
+  #
+  # 2. We wait a certain period of time after we mock
+  #    a node down, so that the buffer period is spent,
+  #    and the node is added to pool.
+
+  describe "start_link" do
+    test "correctly starts a registered GenServer" do
+      start_supervised!({NodePool, name: Foo})
+
+      # Verify Process is running
+      assert Process.whereis(Foo)
+    end
+  end
+
+  describe "get_name/2" do
+    test "creates a new node name if pool is empty" do
+      start_supervised!({NodePool, name: Bar})
```

Comment:
One trick is to name the process after the test:

```suggestion
    test "creates a new node name if pool is empty", config do
      start_supervised!({NodePool, name: config.test})
```

Please change this and the tests below accordingly. :)